src/panels/lovelace/cards/hui-gauge-card.js

```diff
@@ -0,0 +1,182 @@
+          width: calc(var(--base-unit) * 4);
+          height: calc(var(--base-unit) * 2);
+          top: 0%;
+          border-radius:calc(var(--base-unit) * 2.5) calc(var(--base-unit) * 2.5) 0px 0px ;
+        }
+        .gauge-b{
+          z-index: 3;
+          position: absolute;
+          background-color: var(--paper-card-background-color);
+          width: calc(var(--base-unit) * 2.5);
+          height: calc(var(--base-unit) * 1.25);
+          top: calc(var(--base-unit) * 0.75);
+          margin-left: calc(var(--base-unit) * 0.75);
+          margin-right: auto;
+          border-radius: calc(var(--base-unit) * 2.5) calc(var(--base-unit) * 2.5) 0px 0px ;
+        }
+        .gauge-c{
+          z-index: 2;
+          position: absolute;
+          background-color: var(--label-badge-yellow);
+          width: calc(var(--base-unit) * 4);
+          height: calc(var(--base-unit) * 2);
+          top: calc(var(--base-unit) * 2);
+          margin-left: auto;
+          margin-right: auto;
+          border-radius: 0px 0px calc(var(--base-unit) * 2) calc(var(--base-unit) * 2) ;
+          transform-origin: center top;
+          transition: all 1.3s ease-in-out;
+        }
+        .gauge-data{
+          z-index: 4;
+          color: var(--primary-text-color);
+          line-height: calc(var(--base-unit) * 0.3);
+          position: absolute;
+          width: calc(var(--base-unit) * 4);
+          height: calc(var(--base-unit) * 2.1);
+          top: calc(var(--base-unit) * 1.2);
+          margin-left: auto;
+          margin-right: auto;
+          transition: all 1s ease-out;
+        }
+        .gauge-data #percent{
+          font-size: calc(var(--base-unit) * 0.55);
+        }
+        .gauge-data #title{
+          padding-top: calc(var(--base-unit) * 0.15);
+          font-size: calc(var(--base-unit) * 0.30);
+        }
+      </style>
+      <ha-card on-click='_handleClick'>
+        <div class='container'>
+          <div class='gauge-a'></div>
+          <div class='gauge-b'></div>
+          <div class='gauge-c' id='gauge'></div>
+          <div class='gauge-data'><div id='percent'></div><div id='title'></div></div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      _config: Object,
+    };
+  }
+
+  ready() {
+    super.ready();
+    if (this._config) this._buildConfig();
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    this._config = Object.assign({ min: 0, max: 100 }, config);
+    if (!config || !config.entity) throw new Error('Invalid card configuration');
+
+    if (this.$) this._buildConfig();
+    if (this.hass) this._hassChanged(this.hass);
+  }
+
+  _buildConfig() {
+    const entityStateObj = this.hass.states[this._config.entity];
+    const unitOfMeasurement = this._config.unit_of_measurement || entityStateObj.attributes.unit_of_measurement || '';
```

Comment:
You should check that `entityStateObj` exists before accessing its props. Also, should probably be named `stateObj` to be consistent with rest of codebase